### ISSUE TITLE
wallet.AccountInfo is null

### ISSUE THREAD

**Issue by ArtemShe** [darkfriend77/SubstrateNetApi#21]
Hi!
I'm new in crypto.
Trying to use your SDK
Idea its to create wallets and manage them- at least see balances and transfer 'money' between wallets.
As I understood - balance information is placed at `wallet.AccountInfo.AccountData`
Problem that  when wallet is created or loaded -` wallet.AccountInfo is null `
Using different domains 
- Mandala  test network "wss://mandala6.laminar.codes"
- Karura  "wss://karura.api.onfinality.io/public-ws"
 
I debug SDK code and looks like my problem here (class Wallet)
![image](https://user-images.githubusercontent.com/68066163/136535269-2ab343a9-f3b6-45f5-953e-8f2ab22f91d6.png)

So what should I do to have wallet.AccountInfo?

Regards)

**Comment by darkfriend77** [comment]
Hi,

Did you tried with an Account that exists?

**Comment by ArtemShe** [comment]
I'm confused)
According code of class `Wallet ` method `CreateAsync`  that contains  a line
`Account = Account.Build(keyType, privateKey, publicKey);`
this method create account, and account means/equals a wallet.

Could you provide example  of code with your SDK how to create account/wallet from scratch and check balance.

Another interesting point is that looks like `Wallet.CreateAsync` generate public and private key of wallet at client side.
Its also unclear for because
1) usually server return to client "Id" on registration
2) Looks like `Wallet.CreateAsync`  logic not depend on sever/network, but regarding [documentation ](https://wiki.acala.network/karura/get-started/karura-account#address-format) -  base on network is start with specifics symbols.
And this SDK generate public key not regarding this rules.

Regards)

**Comment by darkfriend77** [comment]
That's correct the private key is generated client-side and is encrypted with a seed you provide.

This will not create an account on the blockchain itself for that purpose you will need to create a transaction sending you some tokens at that point your public key will be registered on the chain as an account. maybe that can clarify some of your questions, https://support.polkadot.network/support/solutions/articles/65000168651-what-is-the-existential-deposit-

**Comment by darkfriend77** [comment]
Here is an example of how to create a transaction.

https://github.com/ajuna-network/SubstrateNetApi/blob/c98ab81275d8c5ac6cbd48c00688ab1b8ae6496a/IntegrationTest/LocalNodeTests.cs#L248

As we are currently heavily refactoring the API, you might wait for one a few days then I will provide more code snippets with the most actual monthly release which makes everything much more smooth with the newest API release.

**Comment by RostislavLitovkin** [comment]
Hello, could you please also provide the example how to get the current balance?

**Comment by darkfriend77** [comment]
Sure,
The easiest way is that you create your SDK for your node with this repository. https://github.com/ajuna-network/Ajuna.SDK

Use this step-by-step https://github.com/ajuna-network/Ajuna.SDK#installing-the-template

If you exchange here SystemStorage.Number with SystemStorageAccountInfo your are good to go.
https://github.com/ajuna-network/Ajuna.SDK.Demos/blob/master/Ajuna.SDK.Demos.DirectPolling/Program.cs

**Comment by rahulsharma0856** [comment]
Hello,
As using the transfer function, substrate chain coins are transferring from one wallet to another but only base coin of that chain as per based upon rpc. If we want to transfer the balance of any other token of that chain, what parameter we need to add?  So, According to the token contract address, balance can be transferred from one wallet to another using substrate dotnet.
